From: Mohamad Al-Jaf <mohamadaljaf@gmail.com>
Subject: [PATCH resend 1/2] wusa: Use case insensitive comparison for expressions.
Message-Id: <20220211010607.184056-1-mohamadaljaf@gmail.com>
Date: Thu, 10 Feb 2022 20:06:06 -0500

Some update packages use both lowercase and uppercase names for
expressions.

For example, Microsoft-Windows-MediaFeaturePack-OOB-Package_x64.msu
uses both Runtime and runtime for expressions.

Signed-off-by: Mohamad Al-Jaf <mohamadaljaf@gmail.com>
---
 programs/wusa/main.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/programs/wusa/main.c b/programs/wusa/main.c
index 11111111111..11111111111 100644
--- a/programs/wusa/main.c
+++ b/programs/wusa/main.c
@@ -480,10 +480,10 @@ static WCHAR *lookup_expression(struct assembly_entry *assembly, const WCHAR *ke
 {
     WCHAR path[MAX_PATH];
 
-    if (!wcscmp(key, L"runtime.system32"))
+    if (!wcsicmp(key, L"runtime.system32"))
     {
 #ifdef __x86_64__
-        if (!wcscmp(assembly->identity.architecture, L"x86"))
+        if (!wcsicmp(assembly->identity.architecture, L"x86"))
         {
             GetSystemWow64DirectoryW(path, ARRAY_SIZE(path));
             return strdupW(path);
@@ -492,7 +492,7 @@ static WCHAR *lookup_expression(struct assembly_entry *assembly, const WCHAR *ke
         GetSystemDirectoryW(path, ARRAY_SIZE(path));
         return strdupW(path);
     }
-    if (!wcscmp(key, L"runtime.windows"))
+    if (!wcsicmp(key, L"runtime.windows"))
     {
         GetWindowsDirectoryW(path, ARRAY_SIZE(path));
         return strdupW(path);

-- 
2.35.1

